src/test/java/com/intel/gkl/compression/DeflaterUnitTest.java

```diff
@@ -61,6 +68,120 @@ public void loadLibrary() {
+        intelDeflater.finish();
+        intelDeflater.deflate(compressed,0,compressed.length);
+        intelDeflater.end();
+
+        javaInflater.setInput(compressed,0,compressed.length);
+        javaInflater.inflate(decompressed,0,decompressed.length);
+        javaInflater.end();
+
+        //assert
+        Assert.assertEquals(decompressed,input);
+    }
+
+    @Test(enabled = false, dataProviderClass = CompressionDataProviders.class, dataProvider = "compressionCases", groups = {"long_tests"})
+    public void intelDeflationIntelInflationCompatibilityMaxSizeTest(int level, boolean nowrap) throws DataFormatException {
+
+        //arrange
+        int len = TestingUtils.MAX_ARRAY_SIZE;
+        int outputLen = TestingUtils.MAX_ARRAY_SIZE;
+        if(level == 0)
+            len = nowrap ? TestingUtils.MAX_SINGLE_DEFLATION_SIZE_GZIP : TestingUtils.MAX_SINGLE_DEFLATION_SIZE_NO_GZIP;
+
+        final byte[] input = TestingUtils.generateRandomDNAArray(len);
+        final byte[] compressed = new byte[outputLen];
+        final byte[] decompressed = new byte[len];
+
+        Deflater intelDeflater = new IntelDeflaterFactory().makeDeflater(level, nowrap);
+        Inflater intelInflater = new IntelInflaterFactory().makeInflater(nowrap);
+
+        //act
+        intelDeflater.setInput(input,0,input.length);
+        intelDeflater.finish();
+        intelDeflater.deflate(compressed,0,compressed.length);
+        intelDeflater.end();
+
+        intelInflater.setInput(compressed,0,compressed.length);
+        intelInflater.inflate(decompressed,0,decompressed.length);
+        intelInflater.end();
+
+        //assert
+        Assert.assertEquals(decompressed,input);
+    }
+
+    @Test(enabled = true, dataProviderClass = CompressionDataProviders.class, dataProvider = "compressionCases")
+    public void intelDeflationJavaInflationCompatibilityTest(int level, boolean nowrap) throws DataFormatException {
+        //arrange
+        final int len = TestingUtils.SIZE_4MB;
+        final int outputLen = (int) (1.2*TestingUtils.SIZE_4MB);
+
+        final byte[] input = TestingUtils.generateRandomDNAArray(len);
+        final byte[] compressed = new byte[outputLen];
+        final byte[] decompressed = new byte[len];
+
+        Deflater intelDeflater = new IntelDeflaterFactory().makeDeflater(level, nowrap);
+        Inflater javaInflater = new Inflater(nowrap);
+
+        //act
+        intelDeflater.setInput(input,0,input.length);
+        intelDeflater.finish();
+        intelDeflater.deflate(compressed,0,compressed.length);
+        intelDeflater.end();
+
+        javaInflater.setInput(compressed,0,compressed.length);
+        javaInflater.inflate(decompressed,0,decompressed.length);
+        javaInflater.end();
+
+        //assert
+        Assert.assertEquals(decompressed,input);
+    }
+
+    @Test(enabled = true, dataProviderClass = CompressionDataProviders.class, dataProvider = "compressionCases")
```

Comment:
Noted to be currently failing with Compression Level 2 and Gzip set to false unable to uncompressing the data.

Reply:
Not valid. Gzip set to false is currently excluded from testing